tpcp/_dataset.py

```diff
@@ -333,6 +373,30 @@ def assert_is_single(self, groupby_cols: Optional[Union[str, List[str]]], proper
                 f"is only a single combination of the columns {groupby_cols} left in a data subset,"
             )
 
+    def is_single_datapoint(self) -> bool:
+        """Return True if index contains only one row.
+
+        This is independent of the groupby settings.
+        This means it is equivalent to `is_single_group` if no grouping is given.
+        """
+        return self.is_single(None)
+
+    def assert_is_single_datapoint(self, property_name) -> None:
+        """Raise error if index does contain more than one row.
+
+        Parameters
+        ----------
+        property_name
+            Name of the property this check is used in.
+            Used to format the error message.
+
+        """
+        if not self.is_single_datapoint():
+            raise ValueError(
+                f"The attribute `{property_name}` of dataset {self.__class__.__name__} can only be accessed if there "
+                f"is only a single row left in the data subset index."
```

Comment:
single row (i.e. single datapoint)

and maybe at the end " independent of the groupby level"